SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java

```diff
@@ -156,6 +158,8 @@ int upload(boolean isFinished) throws IOException {
           throw new RuntimeException(ioe);
         }
       });
+    } catch (UncheckedIOException|NoSuchFileException nsfe) {
+      LOG.debug("Parent file {} did not exist, skipping", directory);
```

Comment:
This is because some directories we scanned contain tmp files, so even though we checked for the directory directly above this, it's possible it got deleted out from under us before we started the filesystem walk